dynawo/sources/Common/DYNClone.hpp

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file DYNClone.hpp
+ * @brief clone shared pointer
+ */
+#ifndef COMMON_DYNCLONE_HPP_
```

Comment:
why HPP instead of H as this is a class declaration?

Reply:
HPP is used for template classes in Dynawo

Reply:
nope, .h is used for class declaration and hpp for implementation of template methods when it is too large to be in the h file

Reply:
In that case, it is an implemention

Reply:
I'll let you  double check, when the template method implementation is within the class declaration it is a .h (PARParametersSet.h, DYNGraph.h ...)

Reply:
For a template method for a non template class yes, but clone is a template method (no class)

Reply:
Ah! this is what I missed initially, thanks.